File: vt3rta-cpp-processing-optimization/repository_before/ReportGenerator.cpp
#include "ReportGenerator.h"
#include "Printer.h"
#include <iostream>

void ReportGenerator::generate(const std::vector<Record>& records,
                               const std::vector<std::string>& names) {
    std::string report = "";

    for (size_t i = 0; i < records.size(); i++) {
        report = report + buildLine(records[i], names);
    }

    Printer printer;
    printer.print(report);

    analyze(report);
}

std::string ReportGenerator::buildLine(const Record& record,
                                       const std::vector<std::string>& names) {
    std::string line = "";

    for (size_t i = 0; i < names.size(); i++) {
        if (names[i] == record.name) {
            line = line + std::to_string(record.id)
                        + ":" + record.name + "\n";
        }
    }

    return line;
}

void ReportGenerator::analyze(const std::string& report) {
    int count = 0;

    for (size_t i = 0; i < report.size(); i++) {
        for (size_t j = 0; j < report.size(); j++) {
            if (std::string(1, report[i]) ==
                std::string(1, report[j])) {
                count++;
            }
        }
    }

    if (count > 0) {
        std::cout << "Analysis count: " << count << std::endl;
    }
}

File: 00mnay-java-batch-style-optimization/repository_before/ReportGenerator.java
import java.util.List;

public class ReportGenerator {

    public void generate(List<User> users, List<String> names) {
        String report = "";

        for (int i = 0; i < users.size(); i++) {
            report = report + buildLine(users.get(i), names);
        }

        Printer printer = new Printer();
        printer.print(report);

        analyze(report);
    }

    private String buildLine(User user, List<String> names) {
        String line = "";

        for (int i = 0; i < names.size(); i++) {
            if (names.get(i).equals(user.name)) {
                line = line + user.id + ":" + user.name + "\n";
            }
        }

        return line;
    }

    private void analyze(String report) {
        int count = 0;

        for (int i = 0; i < report.length(); i++) {
            for (int j = 0; j < report.length(); j++) {
                if (("" + report.charAt(i)).equals("" + report.charAt(j))) {
                    count++;
                }
            }
        }

        if (count > 0) {
            System.out.println("Analysis count: " + count);
        }
    }
}
